Data/ByteString/Lazy.hs

```diff
@@ -419,7 +417,7 @@ append = mappend
 -- | /O(n)/ 'map' @f xs@ is the ByteString obtained by applying @f@ to each
 -- element of @xs@.
 map :: (Word8 -> Word8) -> ByteString -> ByteString
-map f s = go s
+map f = go
```

Comment:
I have to say, I'm not totally convinced by all of these eta contractions. Sure some are fine, but in some cases ghc needs to eta-expand again to get efficient code and if it doesn't then we get terrible performance slowdowns (see e.g. the example of bytestring builder's `empty` function which was defined as `id` rather than `\x -> x`).